src/rx5808-pro-diversity/state.cpp

```diff
@@ -0,0 +1,112 @@
+#include <stddef.h>
+
+#include "state.h"
+
+#include "state_screensaver.h"
+#include "state_search.h"
+#include "state_bandscan.h"
+#include "state_menu.h"
+#include "state_settings.h"
+#include "state_settings_rssi.h"
+
+#include "ui.h"
+#include "buttons.h"
+
+#include "timer.h"
+
+
+void *operator new(size_t size, void *ptr){
+  return ptr;
+}
+
+#define MAX(a, b) (a > b ? a : b)
+#define STATE_BUFFER_SIZE \
+    MAX(sizeof(ScreensaverStateHandler), \
+    MAX(sizeof(SearchStateHandler), \
+    MAX(sizeof(BandScanStateHandler), \
+    MAX(sizeof(MenuStateHandler), \
+    MAX(sizeof(SettingsStateHandler), \
+        sizeof(SettingsRssiStateHandler) \
+    )))))
+;
+
+namespace StateMachine {
+    static void onButtonChange(Button button, Buttons::PressType pressType);
+    static StateHandler *getStateHandler(State stateType);
+
+
+    static uint8_t stateBuffer[STATE_BUFFER_SIZE];
```

Comment:
I cant remember if you were having RAM issues or program space issues. I mostly focused on areas that may affect RAM.

I did see these...
rssiData[CHANNELS_SIZE]
rssiALast[RECEIVER_LAST_DATA_SIZE]
rssiBLast[RECEIVER_LAST_DATA_SIZE]

48+24+24 = 96 bytes which is not too bad considering.